kubernetes/upgrade/checks_test.go

```diff
@@ -149,7 +149,161 @@ func TestK8sComponentRemovedItemsWithError(t *testing.T) {
+				"--bind-address=0.0.0.0",
+				"--insecure-port=0",
+				"--feature-gates=ExpandCSIVolumes=true,StatefulSetMinReadySeconds=true",
+				"--master-service-namespace=default",
+			},
+		},
+		k8s.ControllerManagerID: {
+			cliFlags: []string{
+				"/usr/local/bin/kube-controller-manager",
+				"--bind-address=0.0.0.0",
+				"--insecure-port=0",
+				"--feature-gates=ExpandCSIVolumes=true,StatefulSetMinReadySeconds=true",
+				"--pod-eviction-timeout=100s",
+				"--enable-taint-manager",
+			},
+		},
+		k8s.SchedulerID: {
+			cliFlags: []string{
+				"/usr/local/bin/kube-scheduler",
+				"--bind-address=0.0.0.0",
+				"--insecure-port=0",
+				"--feature-gates=ExpandCSIVolumes=true,StatefulSetMinReadySeconds=true",
+			},
+		},
+	}
+
+	expected := upgrade.ComponentRemovedItemsError{
+		CLIFlags: []upgrade.ComponentItem{
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-apiserver",
+				Value:     "master-service-namespace",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-controller-manager",
+				Value:     "enable-taint-manager",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-controller-manager",
+				Value:     "pod-eviction-timeout",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kubelet",
+				Value:     "container-runtime",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kubelet",
+				Value:     "master-service-namespace",
+			},
+			{
+				Node:      "10.5.0.3",
+				Component: "kubelet",
+				Value:     "container-runtime",
+			},
+			{
+				Node:      "10.5.0.3",
+				Component: "kubelet",
+				Value:     "master-service-namespace",
+			},
+		},
+		FeatureGates: []upgrade.ComponentItem{
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-apiserver",
+				Value:     "ExpandCSIVolumes",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-apiserver",
+				Value:     "StatefulSetMinReadySeconds",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-controller-manager",
+				Value:     "ExpandCSIVolumes",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-controller-manager",
+				Value:     "StatefulSetMinReadySeconds",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-scheduler",
+				Value:     "ExpandCSIVolumes",
+			},
+			{
+				Node:      "10.5.0.2",
+				Component: "kube-scheduler",
+				Value:     "StatefulSetMinReadySeconds",
+			},
+		},
+	}
+
+	for _, id := range []string{k8s.APIServerID, k8s.ControllerManagerID, k8s.SchedulerID} {
+		cfg := k8s.NewStaticPod(k8s.NamespaceName, id)
+		cfg.TypedSpec().Pod = map[string]interface{}{
+			"spec": map[string]interface{}{
+				"containers": []interface{}{
+					map[string]interface{}{
+						"command": checkData[id].cliFlags,
+					},
+				},
+			},
+		}
+
+		require.NoError(t, resourceState.Create(ctx, cfg))
+	}
+
+	cfg := k8s.NewKubeletSpec(k8s.NamespaceName, k8s.KubeletID)
+	cfg.TypedSpec().Args = []string{
+		"--container-runtime=containerd",
```

Comment:
``--container-runtime` and `--container-runtime-endpoint`makes sure we are checking the exact value for a flag key and not just a prefix